nickbeaird/pom-elements
# Add docs github action for PRs

Add the ability to generate docs with each merge into master. This adds actions to both simplify the github checks per push, as well as add new functionality to merges.  

The Github Actions are paid for in minutes for deployment of services. The cost to using Poetry for dependency management with this project may have added additional time. I was previously checking that the installation of all packages was happening with each commit. I removed the need to check for dependency install, and added only pre-commit hooks. This seems to run faster. 

For the merge into the master branch, we are now checking that dependencies install correctly, and publishing the now public docs. The docs are being hosted in Github pages. The process is simple, and the only downside is that you cannot madate that docs are private on a private repo. It's public or nothing....or at least for this project. We will see what Github does in the future. 

QA:
- Run pre-commit checks locally. 
- Make a change to the main pom_elements folder, and check that it makes a docs change. Then push and merge the branch. The docs should auto-deploy. Poetry should run. 
- Pre-commit should run and fail on each push. This would keep anyone from pushing changes if they have not setup their environment correctly. 



GH-13
